### PR TITLE
Add favicon when opening the preview in browser

### PR DESCRIPTION
**Pull request by Parsifa1**
add an favicon for the localhost website for typst-preview.

### CODE REVIEW

File: addons/frontend/favicon.png
Comment (Enter-tainer) [inline addons/frontend/favicon.png:1]: we cannot use this logo. see https://github.com/typst/typst/discussions/375#discussioncomment-5699430. But I think typst-guy should be ok?

Comment (Parsifa1) [reply addons/frontend/favicon.png:1]: > we cannot use this logo. see [typst/typst#375 (comment)](https://github.com/typst/typst/discussions/375#discussioncomment-5699430). But I think typst-guy should be ok?我们不能使用这个标志。请参阅typst/typst#375（评论）。但我认为打字员应该没问题吧？

maybe change its color or do some modify？

>You may create and use modified versions of the Typst logo or its elements (a) for a project that in some way incorporates, depends on, extends, or references Typst and (b) on blogs / social media / ... as long as they are substantial enough to not be confused for the original logo

Comment (Enter-tainer) [reply addons/frontend/favicon.png:1]: Agree. I remember currently typst-preview vscode extension doesn't have a icon either. If we can make a modified version that would be nice.

**Comment by Enter-tainer** [comment]
The new icon LGTM. did you test it locally and see if it works or not?

**Comment by Enter-tainer** [comment]
> The new icon LGTM. did you test it locally and see if it works or not?

image

it doesn't work unfortunately. This may require other changes in typst-preview

**Comment by Myriad-Dreamin** [comment]
> > The new icon LGTM. did you test it locally and see if it works or not?
> 
> image
> 
> it doesn't work unfortunately. This may require other changes in typst-preview

We perhaps must bundle it inside of the frontend like that:
```html
<link rel="icon" type="image/svg+xml" href="data:image/svg+xml,LOVELY_SVG_FAVICON_HERE" />
```

This should ensure the frontend html to be a single file again.

**Comment by Myriad-Dreamin** [comment]
We may use https://cloudconvert.com/svg-to-ico for frontend icon.
And https://cloudconvert.com/svg-to-png for extension icon. [Constraints for rendering SVG images](https://code.visualstudio.com/updates/v1_14#_constraints-for-rendering-svg-images).

**Comment by Parsifa1** [comment]
> > The new icon LGTM. did you test it locally and see if it works or not?新图标 LGTM。你在本地测试一下看是否有效？
> 
> image
> 
> it doesn't work unfortunately. This may require other changes in typst-preview不幸的是它不起作用。这可能需要对打字预览进行其他更改

sorry, i will try and do some test stuff

**Comment by Parsifa1** [comment]
image

Done.

File: addons/frontend/index.html [hunk 1]
```diff
@@ -28,7 +32,10 @@
         let deltaLKlsl = deltaL / 1.0;
         let deltaCkcsc = deltaC / sc;
         let deltaHkhsh = deltaH / sh;
-        let i = deltaLKlsl * deltaLKlsl + deltaCkcsc * deltaCkcsc + deltaHkhsh * deltaHkhsh;
+        let i =
```
Comment (Enter-tainer) [inline addons/frontend/index.html]: please keep other part untouched

Comment (Enter-tainer) [reply addons/frontend/index.html]: okay it's restored already

**Review by Enter-tainer — APPROVED** [review]
LGTM

**Pull request merged**